sympy/series/gruntz.py

```diff
@@ -550,6 +550,18 @@ def mrv_leadterm(e, x):
     #
     w = Dummy("w", positive=True)
     f, logw = rewrite(exps, Omega, x, w)
+    from sympy.core.power import Pow
+    li = []
+
+    for terms in f.atoms(Pow):
+        if terms.has(x):
+            b = terms.base
+            e = terms.exp
+            li.append((terms, exp(e*log(b))))
+
+    for a, b in li:
+        f = f.xreplace({a: b})
+
```

Comment:
Hey @arnabnandikgp @oscarbenjamin I did a thorough review. The approach here is correct. Just one question
1) So the gruntz thesis doesn't talk about any simplification being required after the rewrite, so just not sure we need a simplification on the rewritten function `f`
2) But the thesis talks about the above, in section 3.3 for the initial function `e`

So I think if we just shift the block to the top of the function and use it for `e` rather than `f`, that might look better. I tried it out and we have the same results.
```
>>> from sympy import *
>>> x = Symbol('x', real=True, positive=True)
>>> limit((log(E + 1/x) - 1)**(1 - sqrt(E + 1/x)), x, oo)
oo
>>> limit((1 - 1/x)**x*(log(1 - 1/x) + 1/(x*(1 - 1/x))), x, 1, dir='+')
1
>>> gruntz(y*3*sin(I)*x, x, oo)
oo*I*y
```
I think a more easier version of the above block could be 
```
e = e.replace(lambda f: f.is_Pow and f.exp.has(x), lambda f: exp(log(f.base)*f.exp))
```
and this can be used at the top.

Reply:
Also we had this limit not working 
```
limit((log(E + 1/x) - 1)**(1 - sqrt(E + 1/x)), x, oo)
```
but some close relatives of it were working even before the change through a different branch
```
>>> limit((log(E + 1/x) )**(1 - sqrt(E + 1/x)), x, oo)
1
>>> limit((log(E + 1/x) - 1)**(- sqrt(E + 1/x)), x, oo)
oo
```
I think these should also be added to the tests as a control.

Reply:
Thanks for reviewing this pull request. 
> So I think if we just shift the block to the top of the function and use it for e rather than f, that might look better. I tried it out and we have the same results.

I tried this at first too and was stuck at it for quite some time as I wasn't able to get the following test to work,
https://github.com/sympy/sympy/blob/3a812204052176854bdfc1ea9d12097c3f9307ca/sympy/series/tests/test_limits.py#L836-L838
the above test keeps running into a recursion error, the reason being due to routines in `rewrite` which somehow complicate terms in a number of intermediate steps which I had to manually compute to verify, though later I realised if the pow to exp conversion is done after the rewrite operation things get going peacefully. Though it ought to have been done before rewrite, but I guess since these complications arise during rewrite we can safely proceed with doing this conversion after it.
> I think a more easier version of the above block could be

yes, we can add this instead of the code block.

Reply:
Okay I have one question.
If you say it's just one test that doesn't allow us to refactor `e` and instead we would have to refactor `f` .... I would still like if we could refactor `e` eventually. 

Now if it's just one test, let's add a TODO (and we can also open an issue to keep track) that we don't really expect a refactoring after the rewrite and we would like shift the refactoring for `e` once we can look into the troubling test. Maybe after that we would be ready